toolvoice/scripts/inference/gemini_live_infer.py

```
@@ -0,0 +1,351 @@
+        return parsed
+    except Exception:
+        pass
+
+    return [x.strip() for x in s.split(",") if x.strip()]
+
+
+def sanitize_tool_name(name: str) -> str:
+    return re.sub(r"[^a-zA-Z0-9_-]", "_", name)
+
+
+def normalize_to_gemini_tools(raw_tools: Union[List[Any], Dict[str, Any], List[str]]) -> List[Dict[str, Any]]:
```

Ruff (0.16.1)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)